### PR TITLE
Image Fixes

### PR DESCRIPTION
**Pull request by freddyaboulton**
## Description

Closes: #6433

* The upload event was not triggering at all
* The paste-from-clipboard was not triggering an upload event
* The clear button was not triggering a change event
* The change event was triggering infinitely
* Uploaded images were not preserving their original names

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: .config/playwright.config.js [hunk 1]
```diff
@@ -1,7 +1,12 @@
 export default {
 	use: {
 		screenshot: "only-on-failure",
-		trace: "retain-on-failure"
+		trace: "retain-on-failure",
+		permissions: ["clipboard-read", "clipboard-write"],
+		bypassCSP: true,
+		launchOptions: {
```
Comment (freddyaboulton) [inline .config/playwright.config.js:7]: Needed otherwise you get CORS errors

**Comment by freddyaboulton** [comment]
I have to afk but I figured I would open for review since the functional tests passed. Will fix the backend lint when I'm back.

**Comment by abidlabs** [comment]
Awesome PR @freddyaboulton! 

The only issue I noticed is that pasting an image from the clipboard triggers the `.upload()` event twice. Besides that, LGTM!

Test code:

```py
import gradio as gr

with gr.Blocks() as demo:
    image_input = gr.Image()
    textbox = gr.Textbox()

    def upload(image):
        print("upload")
        return "upload"

    def change(image):
        print("change")
        return "change"
        
    image_input.upload(upload, inputs=[image_input], outputs=textbox)
    image_input.change(change, inputs=[image_input], outputs=textbox)

demo.launch()
```

Btw, separately from this PR, I noticed that if you have an image already uploaded, and you upload another image, the progress bar appears *above* the old image instead of clearing the old image first.

File: js/app/test/image_component_events.spec.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+			await fetch(
+				`https://gradio-builds.s3.amazonaws.com/assets/PDFDisplay.png`
+			)
+		).blob();
+		navigator.clipboard.write([new ClipboardItem({ [blob.type]: blob })]);
+	});
+
+	// Not sure if there's a better way to click the clipboard button
+	await page.pause();
+	await page.getByRole("button").nth(2).click();
```
Comment (pngwn) [inline js/app/test/image_component_events.spec.ts:58]: Give the button a data test id.

Comment (freddyaboulton) [reply js/app/test/image_component_events.spec.ts]: Hard to give the button a test id without adding an extra prop to IconButton. Using the label instead and I think it's working! Thanks

**Comment by freddyaboulton** [comment]
Thanks for the review @abidlabs ! The two upload event issue is fixed! Also fixed the issue with the image not being cleared out if you upload another one.

**Review by abidlabs — APPROVED** [review]
Good stuff!

**Comment by freddyaboulton** [comment]
Thanks for the reviews @abidlabs @pngwn !

**Pull request merged**